lib/src/core/room.dart

```diff
@@ -174,13 +204,65 @@ class Room extends DisposableChangeNotifier with EventsEmittable<RoomEvent> {
       );
     }
 
-    await engine.connect(
-      url,
-      token,
-      connectOptions: connectOptions,
-      roomOptions: roomOptions,
-      fastConnectOptions: fastConnectOptions,
-    );
+    if (_regionUrlProvider?.getServerUrl().toString() != url) {
+      _regionUrl = null;
+      _regionUrlProvider = null;
+    }
+    if (isCloudUrl(Uri.parse(url))) {
+      if (_regionUrlProvider == null) {
+        _regionUrlProvider = RegionUrlProvider(url: url, token: token);
+      } else {
+        _regionUrlProvider?.updateToken(token);
+      }
+      // trigger the first fetch without waiting for a response
+      // if initial connection fails, this will speed up picking regional url
+      // on subsequent runs
+      try {
+        await _regionUrlProvider?.fetchRegionSettings();
```

Comment:
the comment says `without waiting for a response`, but the code `await`s . 
I think we can remove the `await`